sdk/search/search-documents/test/public/typeDefinitions.ts

```diff
@@ -1,54 +1,667 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
```

Comment:
Could you tell me a little more about what you're validating here? I mean I get that we're testing type equality of sorts but what benefit does this give you / what bugs will this catch that are not already handled by the type system?

Reply:
Since these types have been changed to be calculated from the corresponding `Known` extensible enum, The API surface we export stays current through regenerations. This is acceptable now where it wasn't before because tsserver now expands these types in LSP type hints. This test makes it so a regeneration that affects an existing string literal union can't pass CI without having been checked.

Also, since APIView renders these types as `type Foo = \`${KnownFoo}\``, it doesn't indicate whether or not these types have changed. This is a stopgap measure to make sure those changes can't slip through review unnoticed.